File: csharp/exceptions2/Program.cs
using System;

namespace Exceptions
{

    class Program
    {
        public static void Main(string[] args)
        {
            try{
            // using automtically closes opened resources
            using (System.IO.StreamReader sr = new StreamReader(@"abcd.txt"))
            {
                int c = sr.Read();

            }
            }catch(Exception exp){
                Console.WriteLine(exp.Message);
            }

            // System.IO.StreamReader sr=null;
            // try
            // {
            //     sr = new StreamReader(@"abcd.txt");
            //      int c = sr.Read();
            // }
            // catch (Exception exp)
            // {
            //     Console.WriteLine("Exception-->"+exp.Message);
            // }finally{
            //     if (sr!=null){
            //     sr.Close();
            //     }
            // }
        }
    }

}

File: java/exceptions2/Program.java
import java.io.*;
import java.lang.*;

public class Program {
// Try with resource exception
    public static void main(String[] args) {
         try(FileInputStream file = new FileInputStream("abcd.txt")){
           byte[] arr =  file.readAllBytes();
           System.out.println(arr);

        }catch(Exception exp){
            System.out.println("Exception--->"+exp.getMessage());
        }
    }
}


